File: Bubble_Sort/Go/jcla1/bubblesort.go
package bubblesort

func BubbleSort(l []int) []int {
	size := len(l)
	swapped := true

	// Remove the following and change parameter
	// to enable side-effects
	arr := make([]int, size)
	copy(arr, l)

	for swapped {
		swapped = false
		for i := 1; i < size; i++ {
			if arr[i-1] > arr[i] {
				arr[i], arr[i-1] = arr[i-1], arr[i]
				swapped = true
			}
		}
	}

	return arr
}

File: Bubble_Sort/Ruby/majjoha/bubble_sort.rb
def bubble_sort(array)
  n = array.length - 2

  while true do
    swapped = false
    (0..n).each_with_index do |item, i|
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        swapped = true
      end
    end
    break unless swapped
  end
  array
end
